### PR TITLE
[X86][AMX] Support AMX-TRANSPOSE, part 2

### PR DESCRIPTION
**Pull request by phoebewang**
Ref.: https://cdrdv2.intel.com/v1/dl/getContent/671368

### CODE REVIEW

File: llvm/lib/Target/X86/X86LowerAMXType.cpp [hunk 1]
```diff
@@ -275,6 +276,27 @@ std::pair<Value *, Value *> ShapeCalculator::getShape(IntrinsicInst *II,
     Col = II->getArgOperand(1);
     break;
   }
+  case Intrinsic::x86_ttdpbf16ps_internal:
+  case Intrinsic::x86_ttdpfp16ps_internal:
+  case Intrinsic::x86_ttcmmimfp16ps_internal:
+  case Intrinsic::x86_ttcmmrlfp16ps_internal:
+  case Intrinsic::x86_tconjtcmmimfp16ps_internal: {
```
[the remark this answers is not in the document]
Comment (phoebewang) [reply llvm/lib/Target/X86/X86LowerAMXType.cpp]: Done

File: clang/lib/Headers/amxbf16transposeintrin.h [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+///
+/// This intrinsic corresponds to the \c TTDPBF16PS instruction.
+///
+/// \param dst
+///    The destination tile. Max size is 1024 Bytes.
+/// \param a
+///    The 1st source tile. Max size is 1024 Bytes.
+/// \param b
+///    The 2nd source tile. Max size is 1024 Bytes.
+#define _tile_tdpbf16ps(dst, a, b) __builtin_ia32_ttdpbf16ps(dst, a, b)
```
[the remark this answers is not in the document]
Comment (phoebewang) [reply clang/lib/Headers/amxbf16transposeintrin.h]: Done.

File: clang/lib/Headers/amxcomplextransposeintrin.h [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+///
+/// This intrinsic corresponds to the \c TTCMMIMFP16PS instruction.
+///
+/// \param dst
+///    The destination tile. Max size is 1024 Bytes.
+/// \param a
+///    The 1st source tile. Max size is 1024 Bytes.
+/// \param b
+///    The 2nd source tile. Max size is 1024 Bytes.
+#define _tile_tcmmimfp16ps(dst, a, b) __builtin_ia32_ttcmmimfp16ps(dst, a, b)
```
[the remark this answers is not in the document]
Comment (phoebewang) [reply clang/lib/Headers/amxcomplextransposeintrin.h]: Done.

File: clang/lib/Headers/amxcomplextransposeintrin.h [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+///
+/// This intrinsic corresponds to the \c TTCMMIMFP16PS instruction.
+///
+/// \param dst
+///    The destination tile. Max size is 1024 Bytes.
+/// \param a
+///    The 1st source tile. Max size is 1024 Bytes.
+/// \param b
+///    The 2nd source tile. Max size is 1024 Bytes.
+#define _tile_tcmmrlfp16ps(dst, a, b) __builtin_ia32_ttcmmrlfp16ps(dst, a, b)
```
[the remark this answers is not in the document]
Comment (phoebewang) [reply clang/lib/Headers/amxcomplextransposeintrin.h]: Done.

File: clang/lib/Headers/amxcomplextransposeintrin.h [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+/// This intrinsic corresponds to the \c TCONJTCMMIMFP16PS instruction.
+///
+/// \param dst
+///    The destination tile. Max size is 1024 Bytes.
+/// \param a
+///    The 1st source tile. Max size is 1024 Bytes.
+/// \param b
+///    The 2nd source tile. Max size is 1024 Bytes.
+#define _tile_conjtcmmimfp16ps(dst, a, b)                                      \
+  __builtin_ia32_tconjtcmmimfp16ps(dst, a, b)
```
[the remark this answers is not in the document]
Comment (phoebewang) [reply clang/lib/Headers/amxcomplextransposeintrin.h]: Done.

File: clang/lib/Headers/amxcomplextransposeintrin.h [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+/// zero_tileconfig_start()
+/// \endcode
+///
+/// This intrinsic corresponds to the \c TCONJTFP16 instruction.
+///
+/// \param dst
+///    The destination tile. Max size is 1024 Bytes.
+/// \param a
+///    The source tile. Max size is 1024 Bytes.
+#define _tile_conjtfp16(dst, a) __builtin_ia32_tconjtfp16(dst, a)
```
[the remark this answers is not in the document]
Comment (phoebewang) [reply clang/lib/Headers/amxcomplextransposeintrin.h]: Done.

File: clang/lib/Headers/amxfp16transposeintrin.h [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+///
+/// This intrinsic corresponds to the \c TTDPFP16PS instruction.
+///
+/// \param dst
+///    The destination tile. Max size is 1024 Bytes.
+/// \param a
+///    The 1st source tile. Max size is 1024 Bytes.
+/// \param b
+///    The 2nd source tile. Max size is 1024 Bytes.
+#define _tile_tdpfp16ps(dst, a, b) __builtin_ia32_ttdpfp16ps(dst, a, b)
```
[the remark this answers is not in the document]
Comment (phoebewang) [reply clang/lib/Headers/amxfp16transposeintrin.h]: Done.

**Review by fzou1 — APPROVED** [review]
LGTM

**Pull request merged**